### DISCUSSION TITLE
New layout suggestions

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#1083] [Ideas]
I am thinking about adding some new node layouts which I call "single line layouts".
They can be used to align child nodes on one line with their parent node.
It can be useful for example for placing all first level nodes at the common line with the root node.

If I implement all combinations of direction, alignment and side, there are new 18 layouts even without the "automatic" generalization layouts.

<img width="1389" alt="single-line-layouts" src="https://user-images.githubusercontent.com/1558052/230727923-96b4e0b3-634b-4ac0-b553-fbfd84bf06b5.png">

It would make possible arrangements like

<img width="336" alt="single-line-layout-use-case" src="https://user-images.githubusercontent.com/1558052/230727983-9d24242e-445a-4723-8bf5-a153af6181e9.png">

where first-level nodes "1", "2" and "3" are all aligned at the center on the right side of the root

I would like to get your impressions. 
Do you want to use any of the listed layouts or all of them?

Regards,
Dimitry

**Comment by EdoFro** [comment]
I find them useful, for example for ishikawa diagrams or timeline diagrams. I think I will use them.

**Reply by dpolivaev** [reply]
Do you think we need all 18?

**Reply by EdoFro** [reply]
Sorry, I don't know. 
I tried to pick the ones I think we need, but there are reasons for each one of them.

first I thought that we need only the  ".._center.svg" ones, but I realized that if the user wants to align all the child nodes in levels, then you need the left/top and right/bottom ones.

**Reply by macmarrum** [reply]
This is a great idea. I like the proposed ducks-in-a-row layouts.

As to the need for all 18 of them, I've been trying to find benefits of one-sided layouts over both-sided layouts. After some analysis I can see that a one-sided layout enforces a sideAtRoot irrespective of the property set on a node, whereas and both-sided layouts take sideAtRoot into consideration.

Therefore, all 18 layouts make sense to me.

**Comment by EdoFro** [comment]
I think it could be useful that in the menu where the user has to choose it, it has some kind of filters to narrow the one you are looking for.

like this:

![image](https://user-images.githubusercontent.com/35700575/230730454-6db9c4b7-0c9b-44c7-b2e9-9b37fe49bc2f.png)

**Reply by EdoFro** [reply]
like this:

![image](https://user-images.githubusercontent.com/35700575/230730646-eb72c3ab-10f0-4856-9cdc-35e0d37abfda.png)

**Comment by abc16361** [comment]
I like the idea. These layouts really help thinking.

**Comment by Lotus49b** [comment]
While I can't provide specific suggestions, I can say that the new layouts we now have are immensely useful to me.

I think these additional layouts will also be useful, in ways I cannot imagine now until I have a particular need.

Thanks for the innovations!

**Comment by wordmuse** [comment]
So - in this layout, all child nodes would be on the same line as the root, and the grandchildren, grreats, and great-greats, etc. would be at offsets from the child nodes? Am I seeing this correctly?

**Comment by sifran-github** [comment]
I find these line layouts to be useful.
It does allow practical uses.

## UNCOMPLETED THOUGHTS ABOUT LAYOUTS
I had a lot of thoughts about the PROPERTIES (Currently in FP Pre: layoutOrientation, childrenSides, childNodesAlignment) and how to define layouts. I wrote my message 5 times and deleted it...
I wrote about:
I might have some ideas on how to organize them a bit differently?
Maybe allow the user to set them separately instead of choosing a complete layout? @EdoFro suggested adding "layout filtering" but maybe just split the choice of properties?
But this topic is very complicated. @dpolivaev Maybe we can chat about it, it might help me to have some insights.

## QUESTION
But I do have one specific question.
what's the difference between the newly proposed:
![image](https://user-images.githubusercontent.com/5730355/231262230-d90b3302-2fff-4bb5-a944-a988ab107d07.png)
and the current way to define this layout, with base distance to children = 0 (in pre29):
![image](https://user-images.githubusercontent.com/5730355/231261250-2abce065-5f32-4a0d-8ded-0f48515a4f28.png)
?
To be specific, my question is: What is the difference of these layouts in terms of their properties?

**Reply by EdoFro** [reply]
@sifran-github
Today I was also curious and took a look to the code in GitHub. It has its own [branch](https://github.com/freeplane/freeplane/tree/single-line-layouts).
there you can take a look at [the differences](https://github.com/freeplane/freeplane/compare/1.11.x...single-line-layouts)

Maybe there is the answer you are looking for.

**Comment by ldmpub** [comment]
New nodes layout are really great ... I have experienced them this week. There are not so easy to understand but yet very powerful.

Thanks to all the community 👏 and big thanks to @dpolivaev 👏👏👏👏👏👏.

**Comment by wordmuse** [comment]
How do I start playing with the new layouts?

~ Bal

**Reply by ldmpub** [reply]
Hi @wordmuse,

Please find hereafter a short video on how to use / activate new nodes layout in Freeplane.

https://github.com/freeplane/freeplane/assets/1452146/8e008944-f744-4c92-bf28-d696a9562ad4

To put it in a nutshell => open node panel and change layout 👍

**Reply by wordmuse** [reply]
Thank you. I'm playing with it now. I LOVE THIS! Fantastic way to let me organize my thoughts in more meaningful ways! Just love it.

~ Bal

**Comment by jpv-1** [comment]
about the top suggestion trying to look like an Ishikawa.
The central axis is not a series of nodes of degree 1, it is just an extension of the node of degree zero: it is still the node of degree zero. Same for the branches
(sorry, I have a lot of difficulty with English and I have to translate everything you say. Excuse me if what I said has already been said or does not fit here: this is a Google translation).
I work with teenagers and young people. Access to Ishikawa must be simpler (a simple button written Ishikawa sending on a template). If I have to pass them through the window with the layouts, I'm going to lose them.
thank you for all this work that is done here, thank you to all those who cooperate: your work greatly benefits young people who have problems with dyslexia, dysorthography, concentration etc.
texte original : a propos de la suggestion du haut qui essaye de ressembler à un Ishikawa.
L'axe central n'est pas une suite de noeuds de degré 1, c'est juste une extension du noeud de degré zero: c'est encore le noeud de degré zero. Idem pour les branches
Je travaille avec des ados et des jeunes. L'acces à Ishikawa doit être plus simple (un simple bouton écrit Ishikawa envoyant sur un template). Si je dois les faires passer par la fenêtre avec les layouts, je vais les perdre.

**Reply by euu2021** [reply]
Le diagramme d'Ishikawa est difficile à reproduire. A titre d'exemple, j'ai essayé de reproduire ceci (jusqu'à les causes secondaires):
`The Ishikawa diagram is difficult to reproduce. As an example, I tried to reproduce this (until the Secondary causes):`

![image](https://github.com/freeplane/freeplane/assets/77707706/dc309ad1-c85a-4d35-be99-0a10a95843dc)


En utilisant la disposition standard FP, cela ressemble à ceci:
`Using FP standard layout, it looks like this:`

![image](https://github.com/freeplane/freeplane/assets/77707706/e21cce92-8afb-4817-aace-0808791bc0e5)


Maintenant, j'ai placé les nœuds manuellement, en essayant de créer une disposition Ishikawa:
`Now, I placed the nodes manually, trying to make an Ishikawa layout:`

![image](https://github.com/freeplane/freeplane/assets/77707706/9a3c7c95-088d-43b3-972d-112fdc71d206)

**Comment by jpv-1** [comment]
Thank you so much. I have done similar things but with my students, I would really like to find the original organization like with xmind for example (see attachment)(which I do not want to use)
texte original : Merci beaucoup. J'ai fait des choses similaires mais avec mes élèves, je voudrais vraiment retrouver l'organisation originale comme avec xmind par exemple (voir pieces jointes) ( que je ne veux pas utiliser)
![Modèle Ishikawa XMind xmt1](https://github.com/freeplane/freeplane/assets/47791991/0c3578ba-ed9b-47bd-9aec-6b1d0af56b39)
[Modèle Ishikawa XMind.zip](https://github.com/freeplane/freeplane/files/13241101/Modele.Ishikawa.XMind.zip)

**Reply by euu2021** [reply]
Oui, j'ai essayé Xmind, ici, et la disposition Ishikawa fonctionne bien, les nœuds de niveau 2 étant positionnés loin de la colonne vertébrale.
`Yes, I tried Xmind, here, and the Ishikawa layout works well, with the Level 2 nodes getting positioned far from the spine.`

![image](https://github.com/freeplane/freeplane/assets/77707706/d0e9ac68-e0ad-4269-a3fb-2985dedd68eb)

**Comment by jpv-1** [comment]
I also tested with Libreoffice draw, but it is significantly less efficient than freeplane in terms of ergonomics, less intuitive
J'ai aussi fait des essais avec Libreoffice draw, mais il est nettement moins performant que freeplane au niveau ergonomie, moins intuitif

**Comment by jpv-1** [comment]
(from an other discussion) Hi! " Fishbone diagram is the most popular kind of diagram for this purpose." 
Fishbone diagrams are often used in face-to-face group sessions, on a board. Its shape allows you to clearly visualize and anchor a structure in memory. Someone who wants to make a report with this diagram cannot do so even though it is present on paid applications (xmind etc), this is the main problem.
But allowing people who use it to also have the depth of freeplane would be interesting. THANKS
original :
Les diagrammes en arete de poisson sont souvent utilisé en séances de groupe en présentiel, sur tableau.  Sa forme permet de bien visualiser et d'ancrer une structure dans la mémoire. Quelqu'un qui veut faire un compt rendu avec ce diagramme ne peut pas le faire alors qu'il est présent sur des application payante (xmind etc), c'est le probleme principal.
Mais permettre aux gens qui l'utilisent d'avoir en plus la profondeur de freeplane serait intéressant. Merci

**Comment by jpv-1** [comment]
idea : perhaps not make insertion points on the rectangle but consider the axis as part of the contour and allow the insertion of arrows on this "axon"?
[amelioration freeplane.odg](https://github.com/freeplane/freeplane/files/13317665/amelioration.freeplane.odg)

**Comment by Amglnbtr** [comment]
Just a user voice: I currently use xmind, it is great for creating simple mind maps quickly and looks great with minimal effort. I find Fishbone and timeline structures really useful in business setting, such as visualizing product lifecycle or customer journey etc. I hope Freeplane implement these structures. Then only thing holding me back from jumping ships is steep learning curve, which I am planning on undertaking when I have more time.

**Reply by euu2021** [reply]
The timeline layout is interesting, but it's complex. A bit was talked about it at:

https://github.com/freeplane/freeplane/discussions/825#discussioncomment-4149680

https://sourceforge.net/p/freeplane/discussion/758437/thread/49de608ff9/

https://github.com/freeplane/freeplane/discussions/566#discussioncomment-3418658
